docs/how-to-migrate-from-smartlock.md

```diff
@@ -0,0 +1,791 @@
+an app to solicit a credential to simplify new account creation or existing
+account discovery. Hints provide a user identifier and authentication method,
+and may optionally include a display name, profile picture and generated
+password.
+
+### Sending a hint request
+
+In both the Smart Lock for Passwords API and the OpenYOLO API, Intents are used
+to communicate with the user’s credential provider to retrieve an account hint.
+
+With Smart Lock for Passwords, a hint request for an email, phone number or
+Google Sign-in based account would be constructed and dispatched as follows:
+
+```java
+void sendSmartLockHintRequest() {
+  PendingIntent hintPickerIntent =
+      Auth.CredentialsApi.getHintPickerIntent(
+          mGoogleApiClient,
+          new HintRequest.Builder()
+              .setEmailAddressIdentifierSupported(true)
+              .setPhoneNumberIdentifierSupported(true)
+              .setAccountTypes(IdentityProviders.GOOGLE)
+              .setIdTokenRequested(true)
+              .setServerClientId(MY_GOOGLE_CLIENT_ID)
+              .setIdTokenNonce(“nonceValue”)
+              .set
+              .build());
+
+  startIntentSenderForResult(
+      hintPickerIntent.getIntentSender(),
+      RC_HINT,
+      null, // fillInIntent
+      0, // flagsMask
+      0, // flagsValues
+      0); // extraFlags
+}
+```
+
+This code assumes that a `GoogleApiClient` instance has already been created and
+connected. Such pre-flight initialization is not necessary for OpenYOLO,
+where the initialization can be fully synchronous, and the code changes to the
+following:
+
+```java
+
+private mCredentialClient =
+    CredentialClient.getApplicationBoundInstance(this);
+
+void sendOpenYoloHintRequest() {
+  Map<String, TokenRequestInfo> tokenProviders = new HashMap();
+  tokenProviders.put(
+      "https://accounts.google.com",
+      new TokenRequestInfo.Builder()
+          .setClientId(MY_GOOGLE_CLIENT_ID)
+          .setNonce("randomlyGeneratedNonce")
+          .build());
+
+  Intent hintIntent = mCredentialClient.getHintRetrieveIntent(
+      new HintRetrieveRequest.Builder(
+          AuthenticationMethods.EMAIL,
+          AuthenticationMethods.PHONE,
+          AuthenticationMethods.GOOGLE)
+          .setTokenProviders(tokenProviders)
+          .build());
+
+  startActivityForResult(hintIntent, RC_HINT);
+}
+```
+
+The key differences to note are:
+
+- The `CredentialClient` is easier to create than a `GoogleApiClient` instance:
+  a `Context` reference is all that is required, with no asynchronous
+  initialization.
+
+- Account types from the Smart Lock for Passwords API are now “authentication
+  methods”, with email auth, phone auth and federated authentication methods
+  all treated in a consistent manner.
+
+- A plain `Intent` is used instead of a `PendingIntent` - we have found this to
+  be less confusing for developers, and the additional functionality provided by
+  PendingIntent proved to be unnecessary.
```

Comment:
Not sure if it's necessary to mention this.
Let's keep the guide as simple as possible by removing any unnecessary comment?

Reply:
Sure, removed description of pendingintent vs intent.